integration/msp/rsaca_test.go

```diff
@@ -0,0 +1,467 @@
+		testDir, err = ioutil.TempDir("", "msp")
+		Expect(err).NotTo(HaveOccurred())
+
+		client, err = docker.NewClientFromEnv()
+		Expect(err).NotTo(HaveOccurred())
+
+		network = nwo.New(nwo.BasicEtcdRaft(), testDir, client, StartPort(), components)
+		network.GenerateConfigTree()
+
+		By("manually bootstrapping MSPs with RSA CAs")
+		generateRSACACrypto(network)
+		network.CreateDockerNetwork()
+		sess, err := network.ConfigTxGen(commands.OutputBlock{
+			ChannelID:   network.SystemChannel.Name,
+			Profile:     network.SystemChannel.Profile,
+			ConfigPath:  network.RootDir,
+			OutputBlock: network.OutputBlockPath(network.SystemChannel.Name),
+		})
+		Expect(err).NotTo(HaveOccurred())
+		Eventually(sess, network.EventuallyTimeout).Should(gexec.Exit(0))
+
+		for _, c := range network.Channels {
+			sess, err := network.ConfigTxGen(commands.CreateChannelTx{
+				ChannelID:             c.Name,
+				Profile:               c.Profile,
+				BaseProfile:           c.BaseProfile,
+				ConfigPath:            network.RootDir,
+				OutputCreateChannelTx: network.CreateChannelTxPath(c.Name),
+			})
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(sess, network.EventuallyTimeout).Should(gexec.Exit(0))
+		}
+		network.ConcatenateTLSCACertificates()
+
+		By("starting all processes for fabric")
+		networkRunner := network.NetworkGroupRunner()
+		process = ifrit.Invoke(networkRunner)
+		Eventually(process.Ready(), network.EventuallyTimeout).Should(BeClosed())
+	})
+
+	AfterEach(func() {
+		if process != nil {
+			process.Signal(syscall.SIGTERM)
+			Eventually(process.Wait(), network.EventuallyTimeout).Should(Receive())
+		}
+		if network != nil {
+			network.Cleanup()
+		}
+		os.RemoveAll(testDir)
+	})
+
+	It("executes transactions endorsed with ECDSA signing certs", func() {
+		org1Peer0 := network.Peer("Org1", "peer0")
+		orderer := network.Orderer("orderer")
+
+		chaincode := nwo.Chaincode{
+			Name:            "mycc",
+			Version:         "0.0",
+			Path:            components.Build("github.com/hyperledger/fabric/integration/chaincode/simple/cmd"),
+			Lang:            "binary",
+			PackageFile:     filepath.Join(testDir, "simplecc.tar.gz"),
+			Ctor:            `{"Args":["init","a","100","b","200"]}`,
+			SignaturePolicy: `AND ('Org1MSP.member','Org2MSP.member')`,
+			Sequence:        "1",
+			InitRequired:    true,
+			Label:           "my_prebuilt_chaincode",
+		}
+
+		network.CreateAndJoinChannels(orderer)
+		nwo.EnableCapabilities(
+			network,
+			"testchannel",
+			"Application", "V2_0",
+			orderer,
+			network.Peer("Org1", "peer0"),
+			network.Peer("Org2", "peer0"),
+		)
+		nwo.DeployChaincode(network, "testchannel", orderer, chaincode)
+		RunQueryInvokeQuery(network, orderer, org1Peer0, 100)
+	})
+})
+
+// What follows is a bunch of code to build a hand-crafted set of MSPs
+// where the CAs use RSA keys and everything else is ECDSA. It's not
+// pretty but it gets the job done for testing.
+
+func generateRSACACrypto(n *nwo.Network) {
```

Comment:
So I don't understand why we are essentially replicating what cryptogen is doing only with RSA CAs. 
Isn't it simpler to:

1. Run cryptogen
2. Locate all CAs, replace their public key and re-sign them.
3. Locate all certificate, replace their signature with a new signature
?

Reply:
This started out with the assumption that RSA was used throughout the MSP so this code generated an entire MSP tree with RSA. When it became clear that signing certificates could not use RSA, I changed that path to use ECDSA. As you can see, that resulted in the introduction of `generateECKey` and a call to it from one site.

As the comment at the top of the code says, "It's not pretty but it gets the job done for testing." I stand by that. Changing this to walk through all of the crypto directories to re-sign certificates is nearly as complicated as what's here. What's here now works, is self-contained, and isolated to the single test that needs it. I do not believe rewriting it to parse and re-sign certs is worth the effort.